### ISSUE TITLE
Base map layers are not included in the print layout

### ISSUE THREAD

**Issue by elpaso** [g3w-suite/g3w-admin#22]
I know the internals and I understand why it happens. 
But from a user perspective this is a bug.

**Comment by wlorenzetti** [comment]
@elpaso do you have a suggestion about to do it? How you kown, we use GetPrint QGIS-Server method tot get pdf/png map, base map usually are xyz layer. With new QGIS API it could be possible add on fly these layers type?

**Comment by elpaso** [comment]
If we switch to a Django-embedded QgsServer instance this would be perfectly doable, we need to investigate the performances though.

**Comment by lucamanga** [comment]
seeing in the QWC2 side, there's a trick:

```
- If you want to have a background when printing, you can add an 
equivalent layer to the QGIS project and specify it as "printLayer" for 
the respective backgroundLayer in the themes configuration. The idea 
behind this is that you can use a fast WMTS for the viewer and an 
equivalent high-quality WMS for printing. Layers in the QGIS project 
marked as printLayer in the themes configuration are not added to the 
QWC2 layer tree.

Note that background layers added to the QGIS project which are not 
marked as printLayer are treated as normal layers, since there is no way 
to know whether a raster layer is a background layer or something else.

So in short: define your background layers in the themesConfig, remove 
them from the QGIS project, except for those you want to use as print 
layers and mark as printLayer in the themesConfig.
```

it would be nice to have it into G3W, too.

**Comment by wlorenzetti** [comment]
Thanks @lucamanga it could be a solution. 
I didn't find time to think about it, but now that we are using the pypthon QGIS-server API it could do a plugin for to create a layer on the fly with base layers activated on map to add to the qgis project before to send to Print service.
Is not if it could be a solution. 
At the moment in the suite we don't use qgis theming.. but we thinking about it.
